File: backend/ArduinoThermoHygrometer/DTOs/TemperatureDTO.cs
namespace ArduinoThermoHygrometer.Web.DTOs;

public record TemperatureDto
{
    public Guid Id { get; set; } = default!;

    public DateTimeOffset RegisteredAt { get; set; } = default!;

    public decimal Temp { get; set; } = default!;
}

File: api/src/main/java/api/arduinothermohygrometer/dtos/TemperatureDto.java
package api.arduinothermohygrometer.dtos;

import java.time.LocalDateTime;

import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

@Builder
public record TemperatureDto(@NotNull LocalDateTime registeredAt,
                             @DecimalMin("-55.00") @DecimalMax("125.00") Double temp
) {}
